jingle/AArch64Arch_jingle.ml

```diff
@@ -69,7 +69,7 @@ include Arch.MakeArch(struct
         match_kr subs (K k) (K k') >>>
         add_subs [Reg(sr_name r1,r1'); Reg(sr_name r2,r2')]
     | I_LDR(_,r1,r2,kr,_),I_LDR(_,r1',r2',kr',_)
-    | I_STR(_,r1,r2,kr),I_STR(_,r1',r2',kr')
+    | I_STR(_,r1,r2,kr,_),I_STR(_,r1',r2',kr',_)
```

Comment:
**Oups:** I guess one should also either check that the shift extensions match (option 1, or extend the match-and-bind process to shift extensions (option 2, for instance bind the shift amount by some metavariable...) . Option 1. is more simple . With current code we face the risk of discarding shift extensions silently.
Unfortunately, I missed this on previous introductions of shift extensions in jingle, all of them have to be handled now, sorry.
In practive, one have to write (and call :) a `match_shift` function, similar to the `match_kr` function. The simple version of this function will check equality returning `Some subs` when equal, `None` otherwise. The sophisticated version will consider the possibility of `MetaConst.Meta m` and perform the adequate bindings.

Reply:
I've implemented the simple one, but I admit I don't quite understand `Meta` at this point and so it might need a look if you can :)